src/player.cpp

```diff
@@ -6969,8 +6978,17 @@ void player::hardcoded_effects(effect &it)
     } else if (id == "grabbed") {
         blocks_left -= 1;
         dodges_left = 0;
-        // Set ourselves up for removal
-        it.set_duration(0);
+        int zed_number = 0;
+        for( auto &dest : g->m.points_in_radius( pos(), 1, 0 ) ){
+            if (g->mon_at(dest) != -1){
+                zed_number ++;
+            }
+        }
+        if (zed_number > 0){
+            add_effect("grabbed", 2, bp_torso, false, (intense + zed_number) / 2); //If intensity isn't pass the cap, average it with # of zeds
+        } else {
+            remove_effect("grabbed"); //If there are no surrounding enemies, remove the grab
```

Comment:
I think this may be causing the crash. In this function, effects remove themselves with `it.set_duration(0);`. Removing it with `remove_effect` may cause some unexpected behavior, including the crash that happens when a grabbed player has no nearby zeds.